src/geometry/manifold/vector_bundle/basic.lean

```diff
@@ -0,0 +1,231 @@
+  modelled on `F`, the change-of-co-ordinates between two trivializations `e`, `e'` for `E`,
+  considered as charts to `B × F`, is smooth and fibrewise linear, in the sense of belonging to the
+  structure groupoid `smooth_fiberwise_linear`.
+
+* `bundle.total_space.smooth_manifold_with_corners`: A smooth vector bundle is naturally a smooth
+  manifold.
+
+* `vector_bundle_core.smooth_vector_bundle`: If a (topological) `vector_bundle_core` smooth, in the
+  sense of having smooth transition functions, then the vector bundle constructed from it is a
+  smooth vector bundle.
+
+* `bundle.prod.smooth_vector_bundle`: The direct sum of two smooth vector bundles is a smooth vector
+  bundle.
+-/
+
+assert_not_exists mfderiv
+
+open bundle set local_homeomorph
+open_locale manifold bundle
+
+variables {𝕜 B B' F M : Type*} {E : B → Type*}
+
+/-! ### Charted space structure on a fiber bundle -/
+
+section
+variables [topological_space F] [topological_space (total_space E)] [∀ x, topological_space (E x)]
+  {HB : Type*} [topological_space HB]
+  [topological_space B] [charted_space HB B]
+
+/-- A fibre bundle `E` over a base `B` with model fibre `F` is naturally a charted space modelled on
+`B × F`. -/
+instance fiber_bundle.charted_space [fiber_bundle F E] :
+  charted_space (B × F) (total_space E) :=
+{ atlas := (λ e : trivialization F (π E), e.to_local_homeomorph) '' trivialization_atlas F E,
+  chart_at := λ x, (trivialization_at F E x.proj).to_local_homeomorph,
+  mem_chart_source := λ x, (trivialization_at F E x.proj).mem_source.mpr
+    (mem_base_set_trivialization_at F E x.proj),
+  chart_mem_atlas := λ x, mem_image_of_mem _ (trivialization_mem_atlas F E _) }
+
+local attribute [reducible] model_prod
+
+/-- Let `B` be a charted space modelled on `HB`.  Then a fibre bundle `E` over a base `B` with model
+fibre `F` is naturally a charted space modelled on `HB.prod F`. -/
+instance fiber_bundle.charted_space' [fiber_bundle F E] :
+  charted_space (model_prod HB F) (total_space E) :=
+charted_space.comp _ (model_prod B F) _
+
+end
+
+/-! ### Smooth vector bundles -/
+
+variables [nontrivially_normed_field 𝕜] [∀ x, add_comm_monoid (E x)] [∀ x, module 𝕜 (E x)]
+  [normed_add_comm_group F] [normed_space 𝕜 F]
+  [topological_space (total_space E)] [∀ x, topological_space (E x)]
+
+  {EB : Type*} [normed_add_comm_group EB] [normed_space 𝕜 EB]
+  {HB : Type*} [topological_space HB] (IB : model_with_corners 𝕜 EB HB)
+  [topological_space B] [charted_space HB B] [smooth_manifold_with_corners IB B]
+  {EB' : Type*} [normed_add_comm_group EB'] [normed_space 𝕜 EB']
+  {HB' : Type*} [topological_space HB'] (IB' : model_with_corners 𝕜 EB' HB')
+  [topological_space B'] [charted_space HB' B'] [smooth_manifold_with_corners IB' B']
+
+variables (F E) [fiber_bundle F E] [vector_bundle 𝕜 F E]
+
+/-- Class stating that a topological vector bundle is smooth, in the sense of having smooth
+transition functions. -/
```

Comment:
Maybe clarify by showing the explicit variables in the docstring. Something like
```suggestion
/-- When `B` is a smooth manifold with corners with respect to a model `IB` and `E` is a
topological vector bundle over `B` with fibers isomorphic to `F`, then `smooth_vector_bundle F E IB`
registers that the bundle is smooth, in the sense of having smooth transition functions.
This is a mixin, not carrying any new data`. -/
```